src/placeos-driver/queue.cr

```diff
@@ -132,12 +127,20 @@ class PlaceOS::Driver::Queue
   end
 
   protected def queue_task(priority, task)
+    task.apparent_priority = priority
+    name = task.name
+
     if @online
-      @queue.push priority, task
+      @queue.unshift task
+      @queue = @queue.sort { |a, b| a.apparent_priority <=> b.apparent_priority }
```

Comment:
sort should be fine to use a mutating method, but the reject! i can see could cause problems

Reply:
a quick experiment shows otherwise, even after adding synchronisation but the leak is extremely aggressive - exponential even - which is entirely weird given the queue in my test code never grows past size 1

I'll see if I can build a minimal example to create an issue against the standard lib

Reply:
https://github.com/crystal-lang/crystal/issues/10748